src/System.CommandLine.Subsystems.Tests/ValueSubsystemTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using System.CommandLine.Directives;
+using System.CommandLine.Parsing;
+using Xunit;
+
+namespace System.CommandLine.Subsystems.Tests;
+
+public class ValueSubsystemTests
+{
+    [Fact]
+    public void Value_is_always_activated()
+    {
+        CliRootCommand rootCommand = [
+            new CliCommand("x")
+            {
+                new CliOption<string>("--opt1")
+            }];
+        var configuration = new CliConfiguration(rootCommand);
+        var subsystem = new ValueSubsystem();
+        var input = "x --opt1 Kirk";
+        var args = CliParser.SplitCommandLine(input).ToList();
+
+        Subsystem.Initialize(subsystem, configuration, args);
+        var parseResult = CliParser.Parse(rootCommand, args[0], configuration);
+        var isActive = Subsystem.GetIsActivated(subsystem, parseResult);
+
+        isActive.Should().BeTrue();
+    }
+
+    [Fact]
+    public void ValueSubsystem_returns_values_that_are_entered()
+    {
+        var consoleHack = new ConsoleHack().RedirectToBuffer(true);
+        CliOption<int> option1 = new CliOption<int>("--intValue");
+        CliRootCommand rootCommand = [
+            new CliCommand("x")
+            {
+                option1
+            }];
+        var configuration = new CliConfiguration(rootCommand);
+        var pipeline = Pipeline.CreateEmpty();
+        pipeline.Value = new ValueSubsystem();
+        const int expected1 = 42;
+        var input = $"x --intValue {expected1}";
+
+        var parseResult = pipeline.Parse(configuration, input); // assigned for debugging
+        pipeline.Execute(configuration, input, consoleHack);
+
+        pipeline.Value.GetValue<int>(option1).Should().Be(expected1);
+    }
+
+    [Fact]
+    public void ValueSubsystem_returns_default_value_when_no_value_is_entered()
+    {
+        var consoleHack = new ConsoleHack().RedirectToBuffer(true);
+        CliOption<int> option1 = new CliOption<int>("--intValue");
+        CliRootCommand rootCommand = [option1];
+        var configuration = new CliConfiguration(rootCommand);
+        var pipeline = Pipeline.CreateEmpty();
+        pipeline.Value = new ValueSubsystem();
+        pipeline.Value.DefaultValue.Set(option1, 43);
+        const int expected1 = 43;
+        var input = $"";
+
+        var parseResult = pipeline.Parse(configuration, input); // assigned for debugging
+        pipeline.Execute(configuration, input, consoleHack);
```

Comment:
```suggestion

        pipeline.Execute(configuration, "", consoleHack);
```
Removing lines that exist just for debugging and unnecessary assignment

This exists multiple places in this file

Reply:
Quite the onion! This may be one of those situations where cutting into it causes more tears than may be worth it at this point in time :)

One minor thing we could consider doing if it is possible (and I am not sure without looking into parseResult), is if that can be converted into some easily readable string format, we could put that as the because parameter in the assertion so we can get some contextual information at the least without having to actually just run the test in debug mode? Not sure if we can get a nice string with some good information, but that is my most simple idea so far to potentially take a step in the right direction without solving the issue entirely.